client/src/user-context.tsx

```diff
@@ -154,6 +154,27 @@ function setSessionStorageData(data: UserData) {
   }
 }
 
+function handleOffline(data) {
```

Comment:
I don't feel great about these side effects in the swr hook: both because swr will consider any error thrown an indication that the request failed, and will attempt to call the function again - which could be solved with a big try/catch - but I think there's a general code smell here of mixing setting accurate userContext data (ensuring that `isSubscriber` and `offlineSettings` aren't incompatible) with doing things as a result of that data (changing what's saved in localStorage, and disabling the service worker). The latter should exist outside the swr function, in a useEffect hook, possibly even outside of the userContext.